cassandane/Cassandane/Test/NewTestUrl.pm

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/perl
+
+package Cassandane::Test::NewTestUrl;
+use strict;
+use warnings;
+
+use JSON;
+use LWP::UserAgent;
+
+use base qw(Cassandane::Unit::TestCase);
+
+sub new
+{
+    my $class = shift;
+    my $self = $class->SUPER::new(@_);
+    return $self;
+}
+
+sub test_basic
+{
+    my ($self) = @_;
+
+    my $ts = bless {}, 'Cassandane::Unit::TestCase';
+
+    my $string_based_url = $ts->new_test_url("string based");
+    my $code_based_url = $ts->new_test_url(sub {
+        return [
+            201,
+            [],
+            [ "code based" ],
+        ],
+    });
+
+    my $lwp = LWP::UserAgent->new;
+
+    {
+        my $res = $lwp->get($string_based_url->url);
+        $self->assert_str_equals('200', $res->code);
+        $self->assert_str_equals('string based', $res->decoded_content);
+    }
```

Comment:
I like this pattern, of putting each "do a thing, get a result, check it" in its own block, rather than having to reuse $res over and over

Reply:
Yeah I love Test::More subtests!  But it's been a while since I've written much perl that wasn't "Cassandane perl", and for good or ill we're wedded to Test::Unit